wildmatch.go

```diff
@@ -69,6 +83,32 @@ const (
 	escapes = "\\[]*?"
 )
 
+// pathname converts patterns of the form "foo*.txt" to "foo/**/*.txt", thus
+// in-effect allowing the "*" operator to match multiple directory levels at
+// once.
+func pathname(p string) string {
```

Comment:
I think this function can be a bit tough to follow, so it might be worth adding a few comments